app/models/katello/glue/candlepin/pool.rb

```diff
@@ -81,20 +81,28 @@ def backend_data
         self.class.candlepin_data(self.cp_id)
       end
 
+      def stacking_subscription(org_label, stacking_id)
+        subscription = ::Katello::Subscription.find_by_product_id(stacking_id)
+        if subscription.nil?
+          found_product = ::Katello::Resources::Candlepin::Product.find_for_stacking_id(org_label, stacking_id)
+          ::Katello::Subscription.find_by_product_id(found_product['id']) if found_product
+        end
+      end
+
       def import_data
         pool_attributes = {}
         pool_json = self.backend_data
         product_attributes = pool_json["productAttributes"] + pool_json["attributes"]
 
         product_attributes.map { |attr| pool_attributes[attr["name"].underscore.to_sym] = attr["value"] }
 
-        if !pool_json["sourceStackId"].nil?
-          subscription = ::Katello::Subscription.where(:product_id => pool_json["sourceStackId"])
+        if pool_json["sourceStackId"]
+          subscription = stacking_subscription(pool_json['owner']['key'], pool_json["sourceStackId"])
         else
-          subscription = ::Katello::Subscription.where(:cp_id => pool_json["subscriptionId"])
+          subscription = ::Katello::Subscription.find_by_cp_id(pool_json["subscriptionId"])
```

Comment:
I would suggest moving away from [find_by_attributes](http://apidock.com/rails/ActiveRecord/FinderMethods/find_by_attributes) even though they [aren't officially deprecated](http://stackoverflow.com/questions/18547571/is-find-by-x-really-mildly-deprecated) its something we moved away from in the Rails 4 conversion and will make us less ready for Rails 5. ```::Katello::Subscription.find_by cp_id: pool_json["subscriptionId"]``` should work the same